### PR TITLE
Am fetchuit datele din api si le am pus in baza de date din memorie

### PR DESCRIPTION
**Pull request by Andu033**

### CODE REVIEW

**Comment by Andu033** [comment]
Am dat push pe master deci s-a mai pus un commit in PR. Ultimul comit adauga inca o ruta care primeste un parametru si returneaza cartiile care apartin unei clase

File: backend/db/index.js [hunk 1]
```diff
@@ -1,56 +1,18 @@
...
-]
 const db = class {
+  setCards (cards) {
+    this.cards = cards
+  }
   getCards () {
-    return [...cards]
+    return [...this.cards]
+  }
+  getCardsByClass (className) {
```
Comment (ionhoria) [inline backend/db/index.js:20]: Using cards.filter() would be more concise.
You could change the entire getCardsByClass function
https://github.com/ionhoria/cdl-hearthstone/blob/7c525cae362e45b15631dc4088fd85e4965fdc0e/backend/db/index.js#L8-L16
to something like this:
```js
  getCardsByClass (className) {
    return cards.filter(card => card.cardClass == className)
```

File: backend/index.js [hunk 2]
```diff
@@ -1,13 +1,29 @@
...
 const db = require('./db')
+const axios = require('axios')
 
 const stats = {}
 const decks = {}
 const app = express()
+let cards
+axios
+  .get('https://api.hearthstonejson.com/v1/29933/enUS/cards.json')
+  .then(function (response) {
```
Comment (ionhoria) [inline backend/index.js]: Please use arrow functions for coding style consistency.
```suggestion
  .then(response => db.setCards(response.data))
```

File: backend/index.js [hunk 3]
```diff
@@ -1,13 +1,29 @@
...
+    cards = response.data
+    db.setCards(cards)
+  })
+  .catch(function (error) {
```
Comment (ionhoria) [inline backend/index.js]: Please use arrow functions for coding style consistency.

**Comment by Andu033** [comment]
+ ruta pt a adauga deck si obiect in db sa le retina

**Review by ionhoria — CHANGES_REQUESTED** [review]
Experiment with having a constructor or method in the `db` class that fetches the card data upon creation or on demand. This would serve to simulate a database connection and data (pre)fetching. See if it is cleaner or easier to build the app logic this way.
```js
const db = class {
  fetchData () {
    axios.get(...).then(...).catch(...)
  }
...
}
```
Write the commit and pull request messages in English.
Mersi :)

File: backend/index.js (hunk 2 above)
Comment (Andu033) [reply backend/index.js]: You mean lambda ? :)

File: backend/index.js [hunk 4]
```diff
@@ -17,13 +17,22 @@ axios
...
+
 app.get('/cards', (req, res) => {
   res.json(db.getCards())
 })
+
 app.get('/classCards/:className', (req, res) => {
   res.json(db.getCardsByClass(req.query.className))
 })
 
+app.post('/addDeck', (req, res, next) => {
```
Comment (ionhoria) [inline backend/index.js]: You should avoid naming endpoints like this (`/appdeck`). Something along the lines of:
`POST`/`PUT` on `/deck/{deckId}/cards` seems more intuitive to me.
Please validate query param and request body data and handle errors.

File: backend/index.js [hunk 5]
```diff
@@ -17,13 +17,22 @@ axios
...
+  const deck = {}
+  deck[req.query.name] = req.body
```
Comment (ionhoria) [inline backend/index.js]: Consider destructuring `req.body` to make it clear what the deck object properties are.

**Review by ionhoria — CHANGES_REQUESTED** [review]

File: backend/index.js (hunk 2 above)
Comment (ionhoria) [reply backend/index.js]: I mean [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions).

File: backend/index.js (hunk 5 above)
Comment (Andu033) [reply backend/index.js]: req.body will be an array of cards written as Jsons. How i would destruct that?

Comment (ionhoria) [reply backend/index.js]: Disregard my initial suggestion.

You shouldn't need an array of objects in the req.body. The server database already holds the card data. Wouldn't an array of card ids suffice?
```js
[id1, id3, id175]
```
Consider commenting your code. Also, this format would indicate that you are expecting an array:
```js
const cardIds = req.body
deck[deckName] = cardIds
```
You still need to make sure that every id received is valid before committing to database.

Comment (Andu033) [reply backend/index.js]: @ionhoria  Check now

**Comment by ionhoria** [comment]
Great job! There's still work to be done and we'll discuss some refactoring later, but this will do for now :)

**Pull request merged**